### PR TITLE
DDPB-2648: Add "not_deputy" option to declaration question

### PR DESCRIPTION
**Pull request by gregtyler**
## Purpose
In [DDPB-2648](https://opgtransform.atlassian.net/browse/DDPB-2648) (and the PR [opg-digi-deps-client#1232](https://github.com/ministryofjustice/opg-digi-deps-client/pull/1232)) we add support for a new `not_deputy` option when submitting a report. This change simply adds support for that new option.

## Approach
I added `not_deputy` to the list of accepted values in `setAgreedBehalfDeputy` and included it in the annotation describing the field.

## Checklist
- [x] I have performed a self-review of my own code
- [x] I have updated documentation (Confluence/GitHub wiki) where relevant
- [x] I have added tests to prove my work, and they follow our [best practices](https://github.com/ministryofjustice/opg-digi-deps-client/wiki/Testing-best-practices)
- [x] I have successfully built my branch to a feature environment
- [x] New and existing unit tests pass locally with my changes (`docker-compose run --rm api sh scripts/apiunittest.sh`)
- [x] There are no Composer security issues (`docker-compose run api php app/console security:check`)
- [ ] The product team have tested these changes

### CODE REVIEW

File: tests/AppBundle/Entity/Report/NdrTest.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        $this->setExpectedException(\InvalidArgumentException::class);
+        $this->ndr->setAgreedBehalfDeputy('BAD_VALUE');
+    }
+
+    public function testValidAgreedBehalfOptions()
+    {
+        $values = ['not_deputy', 'only_deputy', 'more_deputies_behalf', 'more_deputies_not_behalf'];
+        foreach ($values as $value) {
+            $this->ndr->setAgreedBehalfDeputy($value);
+        }
```
Comment (slizzio) [inline tests/AppBundle/Entity/Report/NdrTest.php]: Do we need some assertions here? like setExceptionNotThrown() If thats an option?

File: tests/AppBundle/Entity/Report/ReportTest.php [hunk 2]
```diff
@@ -417,4 +417,18 @@ public function testGetReportTitle($reportType, $expected)
...
+        $this->setExpectedException(\InvalidArgumentException::class);
+        $this->report->setAgreedBehalfDeputy('BAD_VALUE');
+    }
+
+    public function testValidAgreedBehalfOptions()
+    {
+        $values = ['not_deputy', 'only_deputy', 'more_deputies_behalf', 'more_deputies_not_behalf'];
+        foreach ($values as $value) {
+            $this->report->setAgreedBehalfDeputy($value);
+        }
```
Comment (slizzio) [inline tests/AppBundle/Entity/Report/ReportTest.php]: Same here?

File: tests/AppBundle/Entity/Report/NdrTest.php (hunk 1 above)
Comment (slizzio) [reply tests/AppBundle/Entity/Report/NdrTest.php]: or maybe just assert that $this->ndr->getAgreedBehalfDeputy() == $value?

Comment (gregtyler) [reply tests/AppBundle/Entity/Report/NdrTest.php]: Added an assert that the value was successfully set, which ensures it didn't fail silently

File: tests/AppBundle/Entity/Report/ReportTest.php (hunk 2 above)
Comment (gregtyler) [reply tests/AppBundle/Entity/Report/ReportTest.php]: Resolved same as above

**Pull request merged**